specs/account_spec.rb

```diff
@@ -23,149 +46,194 @@
     end
 
     it "Returns the modified balance" do
-      start_balance = 100.0
-      deposit_amount = 25.0
-      account = Bank::Account.new(1337, start_balance)
+      deposit_amount = 25
 
-      updated_balance = account.deposit(deposit_amount)
+      updated_balance = @account.deposit(deposit_amount)
 
-      expected_balance = start_balance + deposit_amount
+      expected_balance = @start_balance + deposit_amount
       updated_balance.must_equal expected_balance
     end
 
     it "Requires a positive deposit amount" do
-      start_balance = 100.0
-      deposit_amount = -25.0
-      account = Bank::Account.new(1337, start_balance)
+      deposit_amount = -25
+
+      proc {
+        @account.deposit(deposit_amount)
+      }.must_raise ArgumentError
+    end
+  end
+
+  describe "Account#add_owner" do
+    it "Adds a new owner when the current owner is nil" do
+      id = 1337
+      balance = 100
+      owner = Bank::Owner.new(id: 12345, first_name: "Alix")
+
+      account = Bank::Account.new(id: id, balance: balance)
+      account.add_owner(owner)
+
+      account.owner.first_name.must_equal owner.first_name
+    end
+
+    it "Does not add a new owner when it is already set" do
+      id = 1337
+      balance = 100
+      owner = Bank::Owner.new(id: 12345, first_name: "Alix")
+      account = Bank::Account.new(id: id, balance: balance, owner: owner)
+
+      another_owner = Bank::Owner.new(id: 12334, first_name: "Hamlet")
 
       proc {
-        account.deposit(deposit_amount)
+        account.add_owner(another_owner)
       }.must_raise ArgumentError
+
+      account.owner.first_name.must_equal owner.first_name
     end
   end
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Wave 2" do
+describe "Wave 2" do
   describe "Account.all" do
+    before do
+      @accounts = Bank::Account.all
+    end
+
     it "Returns an array of all accounts" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Account.all returns an array
-      #   - Everything in the array is an Account
-      #   - The number of accounts is correct
-      #   - The ID and balance of the first and last
-      #       accounts match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
+      # Account.all returns an array
+      @accounts.must_be_instance_of Array
+    end
+
+    it "All items in returned array are Account instances" do
+      @accounts.each { |account| account.must_be_instance_of Bank::Account }
+    end
+
+    it "Returns the correct number of accounts" do
+      number_of_accounts = CSV.read("support/accounts.csv").count
+      @accounts.length.must_equal number_of_accounts
+    end
+
+    it "Returns a complete array of accounts from the csv file" do
+      # The ID and balance of the first and last
+      # accounts match what's in the CSV file
+      first_account = CSV.read("support/accounts.csv").first
```

Comment:
Rather than read the file in several times over many tests.  Maybe read it in once in a `before` block.  It would DRY things up slightly and speed up the testing.